### PR TITLE
Update SDK

### PR DESCRIPTION
**Pull request by wtgodbe**

### CODE REVIEW

**Review by dougbu — APPROVED** [review]
Just blocked by another test that should be quarantined❔

**Comment by wtgodbe** [comment]
Rerunning to see if it's flaky before I quarantine

**Comment by wtgodbe** [comment]
Test `LinkedApplicationWorks` failed both times, with different failure cases each time:

> System.Net.Http.HttpRequestException : No connection could be made because the target machine actively refused it. (localhost:51307)\r\n---- System.Net.Sockets.SocketException : No connection could be made because the target machine actively refused it.

versus:

> Assert.False() Failure\r\nExpected: False\r\nActual:   True
> | [24.856s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Warning: C:\h\w\C0E60A2E\t\c0eaab8d31cf4ae2b0f2ed5890160593\BasicLinkedApp.exe stderr: Failed to create CoreCLR, HRESULT: 0x80004005

Looks like the test has been quarantined before: https://github.com/dotnet/aspnetcore/issues/24755. It's concerning that it now fails consistently after updating the SDK - @davidfowl looks like last time it might have been related to trimming? https://github.com/dotnet/aspnetcore/pull/32166#issuecomment-826955356

@dsplaisted are you aware of any potentially related SDK changes?

**Comment by dsplaisted** [comment]
@agocke @jeffschwMSFT Do you know if something changed recently with linking that could have impacted this test?

**Comment by agocke** [comment]
We certainly wouldn't expect CoreCLR failing to load. Could you gather a log by setting `COREHOST_TRACE=1`?

**Comment by JunTaoLuo** [comment]
Looks like two failure modes are related. In the test run, I see that `No connection could be made because the target machine actively refused it.` occurs since the host being launched failed with the error `ailed to create CoreCLR, HRESULT: 0x80004005` so it's likely all of these failures are caused by the same underlying reason.

@agocke I was able to get some corehost trace logs but I couldn't make much sense of it:
```
| [42.551s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Warning: C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\BasicLinkedApp.exe stderr: Property FX_DEPS_FILE = 
| [42.558s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Warning: C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\BasicLinkedApp.exe stderr: Property TRUSTED_PLATFORM_ASSEMBLIES = C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.IO.MemoryMappedFiles.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Collections.Immutable.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\BasicLinkedApp.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.ComponentModel.Primitives.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.Win32.Primitives.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.Extensions.DependencyInjection.Abstractions.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Collections.Concurrent.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Collections.Specialized.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.Extensions.DependencyInjection.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Collections.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Text.Encodings.Web.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Collections.NonGeneric.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.ComponentModel.TypeConverter.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.ComponentModel.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.Extensions.Primitives.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.Extensions.Logging.Console.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Console.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Net.Security.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Diagnostics.DiagnosticSource.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.AspNetCore.Server.Kestrel.Core.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Diagnostics.StackTrace.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.IO.Compression.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Net.Primitives.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.IO.FileSystem.Watcher.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.Extensions.Hosting.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Linq.Expressions.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Linq.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Memory.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Net.Http.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Net.Sockets.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.ObjectModel.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Private.CoreLib.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Private.Uri.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Reflection.Metadata.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Runtime.CompilerServices.Unsafe.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.AspNetCore.Connections.Abstractions.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Runtime.InteropServices.RuntimeInformation.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Runtime.Numerics.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Security.Claims.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.Extensions.Options.ConfigurationExtensions.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Security.Cryptography.Algorithms.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.Extensions.FileSystemGlobbing.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Security.Cryptography.Cng.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Security.Cryptography.Csp.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Security.Cryptography.Encoding.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Security.Cryptography.Primitives.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Security.Cryptography.X509Certificates.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.Extensions.Logging.Abstractions.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Security.Principal.Windows.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Text.Json.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Web.HttpUtility.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.Extensions.Logging.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.Extensions.Configuration.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.Extensions.Options.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.Extensions.Configuration.Abstractions.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.Extensions.Configuration.Binder.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.Extensions.FileProviders.Physical.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.Extensions.Configuration.CommandLine.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.Extensions.Configuration.EnvironmentVariables.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.Extensions.Configuration.FileExtensions.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.Extensions.FileProviders.Abstractions.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.Extensions.Logging.Configuration.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.Extensions.Hosting.Abstractions.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.IO.Pipelines.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.AspNetCore.Hosting.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.AspNetCore.Hosting.Abstractions.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.AspNetCore.Hosting.Server.Abstractions.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.AspNetCore.Http.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.AspNetCore.Http.Abstractions.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.AspNetCore.Http.Features.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.AspNetCore.Server.Kestrel.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.AspNetCore.WebUtilities.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.Extensions.Features.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\Microsoft.Net.Http.Headers.dll;C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\System.Private.CoreLib.dll
| [42.558s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Warning: C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\BasicLinkedApp.exe stderr: Property NATIVE_DLL_SEARCH_DIRECTORIES = C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\;
| [42.558s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Warning: C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\BasicLinkedApp.exe stderr: Property PLATFORM_RESOURCE_ROOTS = 
| [42.558s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Warning: C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\BasicLinkedApp.exe stderr: Property APP_CONTEXT_BASE_DIRECTORY = C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\
| [42.559s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Warning: C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\BasicLinkedApp.exe stderr: Property APP_CONTEXT_DEPS_FILES = C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\BasicLinkedApp.deps.json
| [42.559s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Warning: C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\BasicLinkedApp.exe stderr: Property PROBING_DIRECTORIES = 
| [42.559s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Warning: C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\BasicLinkedApp.exe stderr: Property System.Resources.ResourceManager.AllowCustomResourceTypes = false
| [42.559s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Warning: C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\BasicLinkedApp.exe stderr: Property RUNTIME_IDENTIFIER = win10-x64
| [42.559s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Warning: C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\BasicLinkedApp.exe stderr: Property System.Runtime.InteropServices.EnableConsumingManagedCodeFromNativeHosting = false
| [42.559s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Warning: C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\BasicLinkedApp.exe stderr: Property System.ComponentModel.TypeConverter.EnableUnsafeBinaryFormatterInDesigntimeLicenseContextSerialization = false
| [42.559s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Warning: C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\BasicLinkedApp.exe stderr: Property System.Runtime.InteropServices.BuiltInComInterop.IsSupported = false
| [42.559s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Warning: C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\BasicLinkedApp.exe stderr: Property System.Runtime.InteropServices.EnableCppCLIHostActivation = false
| [42.559s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Warning: C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\BasicLinkedApp.exe stderr: Property System.StartupHookProvider.IsSupported = false
| [42.559s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Warning: C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\BasicLinkedApp.exe stderr: CoreCLR path = 'C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\coreclr.dll', CoreCLR dir = 'C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\'
| [42.563s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Warning: C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\BasicLinkedApp.exe stderr: Loaded library from C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\coreclr.dll
| [42.589s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Warning: C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\BasicLinkedApp.exe stderr: Failed to create CoreCLR, HRESULT: 0x80004005
| [42.589s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Warning: C:\h\w\A9ED0900\t\a04d9bd7044d42dbbf2f760d4ce91205\BasicLinkedApp.exe stderr: Failed to create CoreCLR, HRESULT: 0x80004005
```

Here's the full log if you want more details: 
[Standard_Console_Output.log](https://github.com/dotnet/aspnetcore/files/6600482/Standard_Console_Output.log)

**Comment by agocke** [comment]
This one's got me stumped -- no idea how trimming could affect this. @vitek-karas Any ideas?

@JunTaoLuo If you comment out `PublishTrimmed` in `BasicLinkedApp` does the problem still repro?

**Comment by JunTaoLuo** [comment]
>If you comment out PublishTrimmed in BasicLinkedApp does the problem still repro?

Guess we'll find out in the next CI run 😄

**Comment by JunTaoLuo** [comment]
@agocke Looks like the problem is resolved when trimming is disabled. I'm going to revert the commit since it's not something we want to actually check in.

**Comment by agocke** [comment]
Yup, I'm cloning the repo now

**Comment by agocke** [comment]
OK found the problem -- it looks like the CLR is failing to start because it's looking for `System.__ComObject` and failing. I bet this is fallout from the work to disable COM support in the CLR @LakshanF

**Comment by LakshanF** [comment]
We disable built-in COM support by default when trimming. The feature switch details for this is [BuiltInComInteropSupport](https://github.com/dotnet/runtime/blob/main/docs/workflow/trimming/feature-switches.md) and can configure to set it to true in your project and see if that makes the test pass.

**Comment by agocke** [comment]
@JunTaoLuo Any chance we can set `<BuiltInComInteropSupport>true</BuiltInComInteropSupport>` in the `BasicLinkedApp` project file and see if the test passes afterwards?

If so, I think the problem is that the test is appropriately failing -- ASP.NET uses COM, but COM is not supported when trimming.

This failure experience is obviously unfortunate though. We'll try to improve that.

**Comment by vitek-karas** [comment]
While it might be true that ASP.NET needs COM (would be interesting to know what for - in the simple web app case), it doesn't explain this failure.
Based on the trace this fails VERY early on - during `coreclr_initialize`. There should be no user code running at that time (only CoreLib is loaded by the time `coreclr_intialize` returns).

In fact this is trivial to repro - just plain console hello world when fully trimmed will fail to start the runtime. I filed https://github.com/dotnet/runtime/issues/53898 to track that.

Setting `<BuiltInComInteropSupport>true</BuiltInComInteropSupport>` fixes the problem.

File: global.json [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {
   "sdk": {
-    "version": "6.0.100-preview.5.21264.3"
+    "version": "6.0.100-preview.6.21301.1"
```
Comment (TanayParikh) [inline global.json:3]:
```suggestion
    "version": "6.0.100-preview.6.21308.9"
```

File: global.json [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
   },
   "tools": {
-    "dotnet": "6.0.100-preview.5.21264.3",
+    "dotnet": "6.0.100-preview.6.21301.1",
```
Comment (TanayParikh) [inline global.json:6]:
```suggestion
    "dotnet": "6.0.100-preview.6.21308.9",
```

**Comment by agocke** [comment]
Not sure about the Span problem, but this is definitely our bug at the moment, and the `BuiltInComInteropSupport` change will be a workaround -- but we need to fix the underlying problem ASAP.

File: src/Hosting/test/FunctionalTests/LinkedApplicationTests.cs [hunk 3]
```diff
@@ -41,8 +41,6 @@ public async Task LinkedApplicationWorks()
                     StatusMessagesEnabled = false
                 };
 
-                deploymentParameters.EnvironmentVariables["COREHOST_TRACE"] = "1";
```
Comment (davidfowl) [inline src/Hosting/test/FunctionalTests/LinkedApplicationTests.cs]: Why?

Comment (TanayParikh) [reply src/Hosting/test/FunctionalTests/LinkedApplicationTests.cs]: I believe that was just added to help debug the CI issues: https://github.com/dotnet/aspnetcore/pull/33187/commits/bb2e26309c069a00756f7fc92f8876d067a8aee5

File: src/Http/Routing/src/Patterns/RoutePatternParser.cs [hunk 4]
```diff
@@ -568,7 +568,7 @@ private string DebuggerToString()
                 }
                 else
                 {
-                    return _template.Substring(0, _index) + "|" + _template.Substring(_index);
+                    return string.Concat(_template.Substring(0, _index), "|", _template.Substring(_index));
```
Comment (pranavkm) [inline src/Http/Routing/src/Patterns/RoutePatternParser.cs:571]: I would think it wants you to do 
```suggestion
                    return string.Concat(_template.AsSpan(0, _index), "|", _template.AsSpan(_index));
```

File: .editorconfig [hunk 5]
```diff
@@ -151,5 +151,8 @@ dotnet_diagnostic.CA1827.severity = suggestion
 dotnet_diagnostic.CA1829.severity = suggestion
 # CA1834: Consider using 'StringBuilder.Append(char)' when applicable
 dotnet_diagnostic.CA1834.severity = suggestion
+# CA1845: Use span-based 'string.Concat'
```
Comment (pranavkm) [inline .editorconfig:154]: FYI @captainsafia I made a small change to your PR

Comment (captainsafia) [reply .editorconfig:154]: Ah I see -- didn't realize these were meant to be ordered.

**Pull request merged**